hana/extract_hana_package.sls

```diff
@@ -0,0 +1,47 @@
+{%- from "hana/map.jinja" import hana with context -%}
+
+{% if hana.hana_archive_file is defined %}
+{% set hana_package = hana.hana_archive_file %}
+{% set hana_extract_dest = hana.hana_extract_dir %}
+
+setup_hana_extract_directory:
+    file.directory:
+      - name: {{ hana_extract_dest }}
+      - mode: 755
+      - makedirs: True
+
+{% if (".ZIP" in hana_package) or (".zip" in hana_package) or (".RAR" in hana_package) or (".rar" in hana_package) %}
+
+extract_hana_archive:
+    archive.extracted:
+      - name: {{ hana_extract_dest }}
+      - enforce_toplevel: False
+      - source: {{ hana_package }}
+
+{% elif (".exe" in hana_package) or (".EXE" in hana_package) %}
```

Comment:
{%- if hana_package.endswith((".exe", ".EXE")) %}

Reply:
any reason you put `if` instead of `elif`? With if I would have to also add `endif` after each condition. With elif looks cleaner I think

Reply:
Yes, the `elif` is fine. I just copy/pasted the previous command to add the `endswith` option